master/buildbot/test/util/integration.py

```diff
@@ -21,16 +21,65 @@
 
 from twisted.internet import defer
 from twisted.internet import reactor
+from twisted.internet import task
 from twisted.trial import unittest
 
 from buildbot.master import BuildMaster
 from buildbot.status.results import SUCCESS
 from buildbot.status.results import statusToString
 from buildbot.test.util import dirs
+from buildbot.schedulers.basic import BaseBasicScheduler
 from buildslave.bot import BuildSlave
 from buildslave.bot import LocalBuildSlave
 
 
+class WaitForStartConsumingSchedulers(object):
+    """
+    startService does not return a deferred. So, when
+    we call yield master.reconfigServiceWithBuildbotConfig
+    we cannot know whether services, like schedulers, have finished their
+    setup.
+    This creates an issue regarding the SingleBranch, AnyBranch schedulers.
+    These schedulers are able to trigger builds only after they have
+    subscribed to the 'new' change event. If we try to start a build with a new change
+    before they subscribed to the event (what can happen as startService does not return a deferred),
+    the build will never start, and the test will timeout.
+    This class implements a poller to ensure that these schedulers have subscribed to the event,
+    before any test tries to create a build with a new change.
+    """
+    POLL_INTERVAL_SEC = 1
```

Comment:
My point of view is the following:
- this commit should be merged like that, a.s.a.p, to stop disturbing buildbot developers with unit test false positives. In each PR, there is a debate on whether the test fails due to the code in the PR, which is not comfortable at all.
- then I propose we work on a solution, like changing BasicBaseScheduler into an asyncService (if it's a ClusteredService, it may be for a reason we have forgotten), in another PR.